src/structures/gdt.rs

```diff
@@ -5,22 +5,61 @@ use crate::structures::tss::TaskStateSegment;
 use crate::PrivilegeLevel;
 use bit_field::BitField;
 use bitflags::bitflags;
+use core::fmt;
 // imports for intra-doc links
 #[cfg(doc)]
 use crate::registers::segmentation::{Segment, CS, SS};
 
+/// 8-byte entry in a descriptor table.
+///
+/// A [`GlobalDescriptorTable`] (or LDT) is an array of these entries, and
+/// [`SegmentSelector`]s index into this array. Each [`Descriptor`] in the table
+/// uses either 1 Entry (if it is a [`UserSegment`](Descriptor::UserSegment)) or
+/// 2 Entries (if it is a [`SystemSegment`](Descriptor::SystemSegment)). This
+/// type exists to give users access to the raw entry bits in a GDT.
+#[derive(Clone)]
+#[repr(transparent)]
+pub struct Entry(u64);
+
+impl Entry {
+    // Create a new Entry from a raw value.
+    const fn new(raw: u64) -> Self {
+        Self(raw)
+    }
+
+    /// The raw bits for this entry. Depending on the [`Descriptor`] type, these
+    /// bits may correspond to those in [`DescriptorFlags`].
+    pub fn raw(&self) -> u64 {
+        self.0
+    }
+}
+
+impl PartialEq for Entry {
+    fn eq(&self, other: &Self) -> bool {
+        self.raw() == other.raw()
+    }
+}
+impl Eq for Entry {}
```

Comment:
We can just `derive` those implementations.

Reply:
Whoops, they have to be manually specified in #381, but you're right they can be derived here.